### ISSUE TITLE
LearningRateMonitor fails to track learning rate when using adafactor optimizer

### ISSUE THREAD

**Issue by toriving** [PyTorchLightning/pytorch-lightning#12067]
## 🐛 Bug

When I used AdamW for transformer training, LearningRateMonitor recorded the learning rate well on the tensorboard.
However, when using Adafactor, it does not track the learning rate.

My guess is that the problem exists in the code of the transformers that provided the Adafactor optimizer.
Transformers set "lr" of Adafactor as initial "lr" and do not update.

https://github.com/huggingface/transformers/blob/de737866f2bbf017ffb18e013528cce58b8e2c5c/src/transformers/optimization.py#L503-L511

That's why it is written to TensorBoard with an initial lr of 0.01.
It is normal when it falls below 0.01 after 10k step by default.
In practice it trains using dropped values, but records in TensorBoard don't.

### To Reproduce

```python
Adafactor(model.parameters(), scale_parameter=True, relative_step=True, warmup_init=False, lr=None)
```
### Expected behavior

In the default setting, after 10k steps, the learning rate should be less than 0.01 recorded in TensorBoard.

### Environment

- PyTorch Lightning Version (e.g., 1.5.0): 1.5.10
- PyTorch Version (e.g., 1.10): 1.11.0
- Python version (e.g., 3.9): 3.8
- transformers : 4.12.2
- OS (e.g., Linux): linux
- CUDA/cuDNN version:
- GPU models and configuration:
- How you installed PyTorch (`conda`, `pip`, source):
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information:

**Comment by rohitgr7** [comment]
hey @toriving !

lr_monitor relies on param_group `lr` to fetch the current learning rate. Might be a compatibility bug on either side.
https://github.com/PyTorchLightning/pytorch-lightning/blob/6fcfd8ce91c5e6d5721243c1d8c7606ee3079460/pytorch_lightning/callbacks/lr_monitor.py#L217-L220

can you share a reproducible script to check this?
https://colab.research.google.com/github/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/bug_report/bug_report_model.ipynb

**Comment by toriving** [comment]
I tried to reproduce, but it wasn't easy.
Because I did it on the larger model and batch size.

The error occurring in the current colab seems to come from what I reported.
As you said, this bug seems that the learning rates key are not compatible with each other.

Transformers' code doesn't update param_group["lr"] anywhere.

https://colab.research.google.com/drive/1wNSMvU7h8F2RxDYSfyaJBYgnC8U2aTer?usp=sharing
